modules/core/src/main/java/org/apache/synapse/mediators/transform/PayloadFactoryMediator.java

```diff
@@ -72,8 +72,9 @@
 
     private List<Argument> pathArgumentList = new ArrayList<Argument>();
     private Pattern pattern = Pattern.compile("\\$(\\d)+");
+	private Pattern ctxPattern = Pattern.compile("\\$(ctx.[^,\"'<>\n}\\]]*)");
```

Comment:
You should probably make this variable "static" to make things more memory efficient.